src/shared/loop-util.c

```diff
@@ -458,17 +458,9 @@ static int loop_device_make_internal(
                         return r;
         }
 
-        if (path) {
-                r = path_make_absolute_cwd(path, &backing_file);
-                if (r < 0)
-                        return r;
-
-                path_simplify(backing_file);
-        } else {
-                r = fd_get_path(fd, &backing_file);
-                if (r < 0)
-                        return r;
-        }
+        r = fd_get_path(fd, &backing_file);
```

Comment:
fd_get_path() is icky. please don't use it only if it really really makes sense, or for debug stuff.

fd_get_path() for open files whose dentry got deleted is weird, as this will be reported suffixed by " (deleted)", and that makes it ambiguous, since people might have an existing dentry that just ahppens to be named that way, and then nobody knows what it measn. moreover we should really propagate the original path here, in case symlinks are involved. this matters because /dev/disk/by-label/ is just so much more useful than /dev/sda7, and loop devices can be backed by other block devices.

Hence, please, if we have the path, let's propagate it, and use fd_get_path() only if we have nothing